### ISSUE TITLE
[5.4] Share method removed

### ISSUE THREAD

**Issue by RonMelkhior** [invisnik/laravel-steam-auth#43]
In Laravel 5.4, the `share` method was deleted and it is mentioned libraries should use `singeleton`.

[Code in question](https://github.com/invisnik/laravel-steam-auth/blob/master/src/SteamServiceProvider.php#L31)

Docs:

https://laravel.com/docs/5.4/upgrade#upgrade-5.4.0

> The `share` method has been removed from the container. This was a legacy method that has not been documented in several years. If you are using this method, you should begin using the `singleton` method instead:

```php
$container->singleton('foo', function () {
    return 'foo';
});
```

**EDIT:** I have submitted a PR (#44) to fix this.

**Comment by BenSherred** [comment]
I have also encountered this issue tonight.

After reading the documentation and testing you can fix this by doing the following:

**1. Open up \vendor\invisnik\laravel-steam-auth\src\SteamServiceProvider,php**

**2. Change the following in the register() function**

`
$this->app['steamauth'] = $this->app->share(
    function () {
        return new SteamAuth();
    }
);
`

to

`
$this->app->singleton('SteamAuth', function($app) {
    return new SteamAuth($app['SteamAuth']);
});`

**Comment by RonMelkhior** [comment]
directly modifying vendor packages is a bad habit. seeing as you should not
upload the vendor directory to your VCS repo.

Also no need for the constructor argument in your code. See my PR for a
good example.

**Comment by ratcho** [comment]
I am suggesting this is what needs to be changed in the package, of course editing the vendor is not suggested but until the package is updated this is a way of fixing it.

And yes I forgot to remove that due to using an example from the Laravel Docs.

**Comment by RonMelkhior** [comment]
The PR was merged and a new release is out ([v2.3.2](https://github.com/invisnik/laravel-steam-auth/releases/tag/v2.3.2)).
